packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts

```diff
@@ -567,11 +574,48 @@ export class AuthResolver {
         userId: user.id,
         workspaceId,
       });
-
-    await this.twoFactorAuthenticationService.validateTwoFactorAuthenticationRequirement(
-      workspace,
-      currentUserWorkspace.twoFactorAuthenticationMethods,
-    );
+      
+    if (isImpersonation === true) {
+      this.logger.log(
+        `Impersonation token exchange attempt for ${email} by ${impersonatorUserId}`,
+        'AuthResolver.getAuthTokensFromLoginToken',
+      );
+      if (workspace.allowImpersonation !== true) {
```

Comment:
We try to avoid nested if structures, and also avoid else {} statements, when possible (because they are hard to follow). Usually it's more elegant to use ifs with an early return / exception thrown